### DISCUSSION TITLE
What is Codespaces loading for?

### DISCUSSION THREAD

**Discussion by LikeDreamwalker** [community/community#75948] [Codespaces] [answered]
### Select Topic Area

Question

### Body

I have been using codespaces for months, due to net issues, it takes me too much time to load codespaces, and I think sometimes it's not loading codespace itself, but other extensions like GitHub copilot:

![image](https://github.com/community/community/assets/45194556/2b866d4f-6dde-4822-a762-6c9808f19265)

What makes me confused is seems like in some scenes, codespaces will always need time and net to load something from remote to my local environment, but what for? I thought codespaces should build a vscode in the browser for sure, and the remote serve will init to execute other tasks, but I will just see copilot keep on downloading things and other extensions keep loading. And finally seems like codespaces have never been started correctly, cause I can't use it for most tasks but as a simple editor.

In my sight, due to network issues, codespaces is just trying to connect server and download some static files, but finally, due to the lost connection of the server, it can' work, and also the downloaded static files are useless. That's strange if that's true, cause I think there must be a better way to tell users codespaces is losing some of the needed connections and need further decision, instead of users just reloading and rebooting the codespaces.

If it's okay please reply me more details about the loading of codespaces, really being interested  and confused in it.

**Comment by shyakachaste** [comment, marked as answer]
It seems like you're experiencing challenges with the performance of Codespaces, particularly related to loading times and network issues. Codespaces is designed to provide a cloud-based development environment, allowing you to code directly in your browser without needing to set up a local development environment. To improve your experience, consider checking your network stability, ensuring a reliable and fast internet connection.If you continue to face difficulties, reaching out to GitHub support is better.

**Reply by LikeDreamwalker** [reply]
Thanks for your reply. I thought that Codespaces was a "page" to send all user tasks to the server, which causes various network issues situation doesn't make sense because it should only be entirely dead or able to be used.
It looks like it is much more different than I think, and I can understand why. Also, if it is convenient, I want to ask, what's the real difference between the vscode.dev and Codespaces except for the server part? Are they actually the same?
Thanks again.

**Comment by shyakachaste** [comment]
vscode.dev is like a quick way to use Visual Studio Code in a web browser. GitHub Codespaces is for making and using online coding spaces connected to your GitHub projects. so they are different.

**Reply by shyakachaste** [reply]
do you have another problem?

**Comment by shyakachaste** [comment]
In addition! With GitHub Codespaces, you can use frameworks and tools without installing them on your computer. It's handy for hassle-free coding and collaborating directly within your GitHub projects.

**Comment by shyakachaste** [comment]
if you have another question , fill free to ask me
